### DISCUSSION TITLE
Weird pre-inlet output

### DISCUSSION THREAD

**Discussion by mojtabaAmiraslanpour** [UvaCsl/HemoCell#94] [General discussion]
Hi,

I have been trying to use manual decomposition of the domain for both pre-inlet and the main domain while using the curvedflow_with_preinlet solver. I get weird results with different decomposition configurations:

![Screenshot 2023-03-01 105318](https://user-images.githubusercontent.com/15649053/222194788-7a5964d0-302a-4ca1-8006-327cbeda94c4.png)
![Screenshot 2023-03-01 110256](https://user-images.githubusercontent.com/15649053/222194790-f643a18d-3583-48d0-bd0a-1cf3af001ebe.png)
![Screenshot 2023-03-01 110307](https://user-images.githubusercontent.com/15649053/222194791-753e9e0f-6600-46be-97e6-68e18d83e212.png)

And as I know I put the pre-inlet slice in the correct position.
Not sure what is happening. Any suggestions?

Best,
Moe

**Comment by mojtabaAmiraslanpour** [comment]
Just tested in two different clusters. One of them works fine, but the other produces these weird contours. I was thinking maybe the libraries are different, so I used the same conda environment on both clusters. Still happening and I don't have any clue what is going on.

**Comment by mojtabaAmiraslanpour** [comment]
Is there any chance that CPU type affects the results?

**Comment by gzavo** [comment]
Sorry I don't have an answer for this one. CPU vendor, etc. of course should not have effect on the functionality.
Since it happens on a boundary, maybe some problem around the MPI calls? I would first look at the MPI versions and implementation, and if the same is used for compilation and execution.
